# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/AudioPacketizer.h
# Commit: c504995adbb4
# Full Hash: c504995adbb459ffb38efbc80540b0e2dcf43ad4
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2020-11-06 04:14:34
# Regressor Bug: 1651745
# File Overlap Count: 1
# Description:
#   Bug 1651745 - Add methods to inspect buffered amount and to clear AudioPacketizer. r=padenot
#   
#   This patch also constifies members where applicable, and adds a unittest for the
#   new methods.
#   
# ==============================================================================

diff -r 605d00735b78 -r c504995adbb4 dom/media/AudioPacketizer.h
--- a/dom/media/AudioPacketizer.h	Thu Nov 05 16:42:20 2020 +0000
+++ b/dom/media/AudioPacketizer.h	Thu Nov 05 16:42:28 2020 +0000
@@ -40,9 +40,8 @@
       : mPacketSize(aPacketSize),
         mChannels(aChannels),
         mReadIndex(0),
-        mWriteIndex(0)
+        mWriteIndex(0),
         // Start off with a single packet
-        ,
         mStorage(new InputType[aPacketSize * aChannels]),
         mLength(aPacketSize * aChannels) {
     MOZ_ASSERT(aPacketSize > 0 && aChannels > 0,
@@ -131,17 +130,25 @@
     mReadIndex += samplesNeeded;
   }
 
+  void Clear() {
+    mReadIndex = 0;
+    mWriteIndex = 0;
+  }
+
   uint32_t PacketsAvailable() const {
     return AvailableSamples() / mChannels / mPacketSize;
   }
 
+  uint32_t FramesAvailable() const { return AvailableSamples() / mChannels; }
+
   bool Empty() const { return mWriteIndex == mReadIndex; }
 
   bool Full() const { return mWriteIndex - mReadIndex == mLength; }
 
-  uint32_t PacketSize() const { return mPacketSize; }
-
-  uint32_t Channels() const { return mChannels; }
+  // Size of one packet of audio, in frames
+  const uint32_t mPacketSize;
+  // Number of channels of the stream flowing through this packetizer
+  const uint32_t mChannels;
 
  private:
   uint32_t ReadIndex() const { return mReadIndex % mLength; }
@@ -152,10 +159,6 @@
 
   uint32_t EmptySlots() const { return mLength - AvailableSamples(); }
 
-  // Size of one packet of audio, in frames
-  uint32_t mPacketSize;
-  // Number of channels of the stream flowing through this packetizer
-  uint32_t mChannels;
   // Two virtual index into the buffer: the read position and the write
   // position.
   uint64_t mReadIndex;